ocaml/testsuite/tests/typing-unique/overwriting_syntax.ml

```diff
@@ -0,0 +1,124 @@
+(* TEST
+   flags += "-extension unique ";
+   flags += "-extension overwriting";
+   expect;
+*)
+
+(******************)
+(* Usage examples *)
+
+let overwrite_tuple = function
+  (a, b) as t -> overwrite_ t with (b, _)
+[%%expect{|
+Line 2, characters 17-41:
+2 |   (a, b) as t -> overwrite_ t with (b, _)
+                     ^^^^^^^^^^^^^^^^^^^^^^^^
+Alert : Overwrite not implemented.
+Uncaught exception: File "ocaml/parsing/location.ml", line 1112, characters 2-8: Assertion failed
+
+|}]
+
+type record = { a : int; b : int }
+
+let overwrite_record = function
+  { a; b } as t -> overwrite_ t with { b = a; a = _ }
```

Comment:
I would also want to test things like `overwrite_ t with { b = a; _ }` and `overwrite_ t with { b = a }`. Similarly for the inline record case.

Reply:
`overwrite_ t with { b = a; _ }` does not work in this PR. The reason is that underscores may only appear in places where expressions may appear.

Reply:
OK. I suppose we don't need that syntax.